demos/draw-hooks.html

```diff
@@ -0,0 +1,231 @@
+                }
+                
+                function enableBlur(u)
+                {
+                    let {ctx} = u;
+                    setCtxProperties(ctx, {filter: 'blur(3px'});
+                }
+
+                function disableBlur(u)
+                {
+                    let {ctx} = u;
+                    ctx.save();
+                    setCtxProperties(ctx, {filter: 'none'});
+                }
+
+                function drawSeriesBoundary(u, s, i)
+                {
+                    let {ctx} = u;
+                    let {width, height} = ctx.canvas;
+                    let {label, color, min:minVal, max:maxVal} = s;
+
+                    let yPos0 = u.valToPos(maxVal, 'y');
+                    let yPos1 = u.valToPos(minVal, 'y');
+
+                    let restoreCtxProperties = setCtxProperties(ctx, {fillStyle:hexToRgbA(color, 0.1)});
+
+                    ctx.fillRect(0, yPos0, width, yPos1-yPos0);
+                    restoreCtxProperties();
+                }
+
+                function drawPointsAsStars(u, s, i)
+                {
+                    let {ctx} = u;
+                    let {width, height} = ctx.canvas;
+                    let {label, color, min:minVal, max:maxVal} = s;
+
+                    let yVals = u.data[i];
+
+                    let restoreCtxProperties = setCtxProperties(ctx, {fillStyle:color});
+
+                    yVals.forEach((val, i) => {
+                        // Can add additional check if visible on Canvas before drawing                        
+                        let cx = u.valToPos(u.data[0][i], 'x'    );
+                        let cy = u.valToPos(val         , s.scale);
+                        drawStar(ctx, cx, cy);
+                        ctx.fill();                      
+                    });
+
+                    restoreCtxProperties();
+                }
+
+                let startRenderTime = null;
+
+                function setStartRenderTime()
+                {
+                    startRenderTime = Date.now();
+                }
+
+                function drawRenderStats(u)
+                {
+                    let {ctx} = u;
+                    let {width:canWidth} = ctx.canvas;
+                    let text = "Time to Draw: " + (Date.now() - startRenderTime) + "ms";
+                    let restoreCtxProperties = setCtxProperties(ctx, {font:'12px Arial', fillStyle:'white'});
+                    
+                    ctx.fillText(text, 6, 18);
+                    restoreCtxProperties();
+                }
+
+                return {
+                    drawClear: [
+                        setStartRenderTime,
+                        drawBackgroundGradient,
+                        enableBlur
+                    ],
+                    drawGrid: [
+                        disableBlur
+                    ],
+                    drawSeries: [
+                        enableBlur,
+                        drawSeriesBoundary,
```

Comment:
i didn't actually realize what this was doing until i disabled it. until that point it looked like it was just part of the a blurred gradient background.

Reply:
Yeah, not a particularly great example. It's more apparent once you zoom in (then you get a bounding box around the visible points).